src/broadcast/participant.rs

```diff
@@ -145,7 +145,13 @@ impl ProtocolParticipant for BroadcastParticipant {
 }
 
 impl InnerProtocolParticipant for BroadcastParticipant {
-    type Context = ();
+    type Context = Vec<ParticipantIdentifier>;
+
+    fn retrieve_context(&self) -> <Self as InnerProtocolParticipant>::Context {
+        let mut ids = self.all_participants();
+        ids.sort();
+        ids
```

Comment:
Sorry for late comment: Broadcast actually never uses context for anything, so we don't necessarily have to spend a bunch of energy implementing this function -- it never gets used. But you already did it, and I don't want to get into something messy by adding `&()` as a valid context again. Maybe add a comment here that says this isn't used and we can make a separate little issue to make a `NoContext` type for broadcast specifically.